src/test/java/com/flightstats/hub/dao/aws/ClusterContentServiceTest.java

```diff
@@ -0,0 +1,96 @@
+package com.flightstats.hub.dao.aws;
+
+import com.flightstats.hub.cluster.LastContentPath;
+import com.flightstats.hub.dao.ChannelService;
+import com.flightstats.hub.dao.ContentDao;
+import com.flightstats.hub.model.*;
+import com.flightstats.hub.util.HubUtils;
+import com.flightstats.hub.util.TimeUtil;
+import lombok.SneakyThrows;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Optional;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ClusterContentServiceTest {
+    @Mock private ClusterContentService ccs;
+    @Mock private ChannelService channelSvc;
+    @Mock private ContentDao mockSpokeWriteDao, mockSpokeReadDao;
+    @Mock private ContentDao mockS3SingleDao, mockS3LargeDao, mockS3BatchDao;
+    @Mock private S3WriteQueue s3WriteQueue;
+    @Mock private ChannelConfig channelConfig;
+    @Mock private LastContentPath lastContentPath;
+    @Mock private HubUtils hubUtils;
+
+    private Content content;
+    private ContentKey contentKey;
+    private String channelName;
+
+    @Before
+    public void initClusterContentService() {
+        channelName = "/testChannel";
+        when(channelSvc.getCachedChannelConfig(channelName)).thenReturn(Optional.of(channelConfig));
+        ccs = new ClusterContentService(channelSvc, mockSpokeWriteDao, mockSpokeReadDao, mockS3SingleDao, mockS3LargeDao, mockS3BatchDao, s3WriteQueue, lastContentPath, hubUtils);
+    }
+
+    @SneakyThrows
+    private void initSimpleContent(boolean isLarge) {
+        byte[] data = "SimpleTest".getBytes();
+        contentKey = new ContentKey(TimeUtil.now(), "someHash");
+        content = Content.builder()
+                .withContentType("text/plain")
+                .withContentKey(contentKey)
+                .withLarge(isLarge)
+                .withData(data).build();
+        when(mockSpokeWriteDao.insert(anyString(), any(Content.class))).thenReturn(contentKey);
+        when(channelConfig.isBatch()).thenReturn(false);
```

Comment:
I know this DRYs a couple tests up a bit, but I think it's unrelated to setting up the content key.  Also, since it's a "given" that ties directly to the assertions, I'd usually prefer to make it explicit in each test so that there's symmetry across all the use cases.

Reply:
Yeah, that's fair.  Moving.